test/distribute-lp-reward.js

```diff
@@ -0,0 +1,328 @@
+	const d = new Date(initialDate.toString())
+	d.setDate(initialDate.getDate() + days)
+	return d
+}
+
+tape('uniswap: distributes reward correctly', function(t) {
+	// 2 weeks between DISTRIBUTION_START & DISTRIBUTION_ENDS
+	const DISTRIBUTION_START = new Date()
+	DISTRIBUTION_START.setDate(DISTRIBUTION_START.getDate() - 15)
+
+	const DISTRIBUTION_ENDS = new Date()
+	DISTRIBUTION_ENDS.setDate(DISTRIBUTION_ENDS.getDate() - 1)
+
+	const addDaysToDistributionStart = addDays(DISTRIBUTION_START)
+
+	const testUniswapTransferEvents = [
+		{
+			time: Math.floor(DISTRIBUTION_START.getTime() / 1000),
+			values: {
+				from: NULL_ADDRESS,
+				value: bigNumberify('10'),
+				to: '0x1'
+			}
+		},
+		{
+			time: Math.floor(addDaysToDistributionStart(1).getTime() / 1000),
+			values: {
+				from: NULL_ADDRESS,
+				value: bigNumberify('10'),
+				to: '0x2'
+			}
+		},
+		{
+			time: Math.floor(addDaysToDistributionStart(1) / 1000),
+			values: {
+				from: NULL_ADDRESS,
+				value: bigNumberify('10'),
+				to: '0x4'
+			}
+		},
+		{
+			time: Math.floor(addDaysToDistributionStart(7).getTime() / 1000),
+			values: {
+				from: NULL_ADDRESS,
+				value: bigNumberify('10'),
+				to: '0x5'
+			}
+		},
+		{
+			time: Math.floor(addDaysToDistributionStart(9).getTime() / 1000),
+			values: {
+				from: NULL_ADDRESS,
+				value: bigNumberify('10'),
+				to: '0x6'
+			}
+		},
+		{
+			time: Math.floor(addDaysToDistributionStart(9).getTime() / 1000),
+			values: {
+				from: '0x4',
+				value: bigNumberify('3'),
+				to: '0x7'
+			}
+		}
+	]
+
+	const liquidityDurationByMultiplier = {
+		1: 10, // 1 week => 0.1% increase
+		2: 20,
+		3: 30
+	}
+
+	const distributionStarts = Math.floor(DISTRIBUTION_START.getTime() / 1000)
+	const distributionEnds = DISTRIBUTION_ENDS.getTime() / 1000
+	const distributionPerSecond = bigNumberify('10')
+	const totalDistribution = distributionPerSecond.mul(bigNumberify(14 * (24 * 60 * 60)))
+
+	const distribution = {}
+
+	getDistributionForPeriodWithMultiplier(
+		distribution,
+		testUniswapTransferEvents,
+		distributionStarts,
+		distributionEnds,
+		distributionPerSecond,
+		parseUniswapTransferEvents,
+		liquidityDurationByMultiplier
+	)
+
+	const expected = {
+		'0x1': '4105529',
+		'0x2': '2994304',
+		'0x4': '2674304',
+		'0x5': '1281860',
+		'0x6': '800000',
+		'0x7': '240000'
+	}
+
+	t.deepEqual(expected, toStringMap(distribution), 'should distribute reward correctly')
+	t.ok(totalDistribution.gte(sum(distribution)), 'distribution does not exceed allowance')
+
+	t.end()
+})
+
+tape('balancer: distributes reward correctly', function(t) {
+	// 2 weeks between DISTRIBUTION_START & DISTRIBUTION_ENDS
+	const DISTRIBUTION_START = new Date()
+	DISTRIBUTION_START.setDate(DISTRIBUTION_START.getDate() - 15)
+
+	const DISTRIBUTION_ENDS = new Date()
+	DISTRIBUTION_ENDS.setDate(DISTRIBUTION_ENDS.getDate() - 1)
+
+	const addDaysToDistributionStart = addDays(DISTRIBUTION_START)
+
+	const testBalancerTransferEvents = [
```

Comment:
- `testBalancerTransferEvents` and `testUniswapTransferEvents` are absolutely the same.
can be declared as one var. 
- maybe one more test is needed - to check the distribution for `uniswap` and `balancer` at the same time as it is in the real usecase

Reply:
```
`testBalancerTransferEvents` and `testUniswapTransferEvents` are absolutely the same.
can be declared as one var.
```
yeah the values are the same but the properties are different